sensitivity-api/src/main/java/com/powsybl/sensitivity/SensitivityComputationResults.java

```diff
@@ -90,61 +111,142 @@ public String getLogs() {
+     */
+    public Collection<SensitivityValue> getSensitivityValuesByFunction(SensitivityFunction function, String contingencyId) {
+        return sensitivityValuesContingencies.get(contingencyId).stream().filter(sensitivityValue -> sensitivityValue.getFactor().getFunction().equals(function)).collect(Collectors.toList());
+    }
+
+    /**
+     * Get a collection of all the sensitivity values associated with given variable
+     * for a specific contingency.
+     *
+     * @param variable sensitivity variable
+     * @param contingencyId the ID of the considered contingency
+     * @return a collection of all the sensitivity values associated with given variable
+     */
+    public Collection<SensitivityValue> getSensitivityValuesByVariable(SensitivityVariable variable, String contingencyId) {
+        return sensitivityValuesContingencies.get(contingencyId).stream().filter(sensitivityValue -> sensitivityValue.getFactor().getVariable().equals(variable)).collect(Collectors.toList());
+    }
+
+    /**
+     * Get the sensitivity value associated with given function and given variable for a specific contingency.
+     *
+     * @param function sensitivity function
+     * @param variable sensitivity variable
+     * @param contingencyId the ID of the considered contingency
+     * @return the sensitivity value associated with given function and given variable
+     */
+    public SensitivityValue getSensitivityValue(SensitivityFunction function, SensitivityVariable variable, String contingencyId) {
+        Optional<SensitivityValue> returnValue;
+        returnValue = sensitivityValuesContingencies.get(contingencyId).stream().filter(sensitivityValue -> sensitivityValue.getFactor().getFunction().equals(function)
+                && sensitivityValue.getFactor().getVariable().equals(variable)).findAny();
+        if (!returnValue.isPresent()) {
+            throw new NoSuchElementException(String.format("Sensitivity value not found for function %s and variable %s at contingencyId %s.", function.getId(), variable.getId(), contingencyId));
+        }
+        return returnValue.get();
+    }
+
+    /**
+     * Get the sensitivity value associated with given factor for a specific contingency
+     *
+     * @param factor sensitivity factor
+     * @param contingencyId the ID of the considered contingency
+     * @return the sensitivity value associated with given function and given variable
+     */
+    public SensitivityValue getSensitivityValue(SensitivityFactor factor, String contingencyId) {
+        Optional<SensitivityValue> returnValue;
+        returnValue = sensitivityValuesContingencies.get(contingencyId).stream().filter(sensitivityValue -> sensitivityValue.getFactor().equals(factor)).findAny();
+        if (!returnValue.isPresent()) {
+            throw new NoSuchElementException(String.format(VALUE_NOT_FOUND, factor.getFunction().getId(), factor.getVariable().getId()));
+        }
+        return returnValue.get();
```

Comment:
Same as above, this code can be simplified. (Code of other similar methods of this class can be simplified as well)